File: backend/src/main/java/citas/entity/Citas.java
package citas.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "Citas")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Citas {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cita")
    private int idCita;

    @ManyToOne
    @JoinColumn(name = "id_paciente", nullable = false, foreignKey = @ForeignKey(name = "fk_citas_paciente"))
    private Pacientes paciente;

    @ManyToOne
    @JoinColumn(name = "id_medico", nullable = false, foreignKey = @ForeignKey(name = "fk_citas_medico"))
    private Medicos medico;

    @Column(name = "fecha_cita", nullable = false)
    private LocalDate fechaCita;

    @Column(name = "hora_cita", nullable = false)
    private LocalTime horaCita;

    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;
}

File: frontend/src/app/models/citas.ts
import { Medico } from "./medico";
import { Paciente } from "./paciente";
export class Citas {
    idCita: number;
    paciente: Paciente; // Relación con Pacientes
    medico: Medico; // Relación con Medicos
    fechaCita: Date; // LocalDate en Java corresponde a Date en TypeScript
    horaCita: string; // LocalTime se maneja como string en formato HH:mm:ss
    descripcion: string;
  
    constructor(
      idCita: number = 0,
      paciente: Paciente = new Paciente(),
      medico: Medico = new Medico(),
      fechaCita: Date = new Date(),
      horaCita: string = '00:00:00',
      descripcion: string = ''
    ) {
      this.idCita = idCita;
      this.paciente = paciente;
      this.medico = medico;
      this.fechaCita = fechaCita;
      this.horaCita = horaCita;
      this.descripcion = descripcion;
    }
  }
  
 